docs/source/_templates/partials/submitting_to_validator.rst

```diff
@@ -16,7 +16,7 @@ prepared the BatchList:
 
 .. code-block:: javascript
 
-    const request = require('request')
+    const request = require('browser-request')
```

Comment:
What is the thinking for using `browser-request` here? This presumes that the JavaScript is being run from within a browser, presumably with `browserify`. On the other hand, `request` is a much more popular library (15x as many downloads), which could be considered a "standard" for Node.js, and should work in the browser as well with `webpack`.

Reply:
@delventhalz Adding `const request = require('request')` to my webpack build produces the following failures (just reproduced on a another machine). Inspecting https://www.npmjs.com/package/request provides no documentation on how to use the library from a browser.  I am happy with any http library that works in the browser environment. The one I ended up using was framework provided so is not a good example. 



```ERROR in ./node_modules/request/lib/har.js
Module not found: Error: Can't resolve 'fs' in '/project/cookie-trader/client/node_modules/request/lib'
 @ ./node_modules/request/lib/har.js 3:9-22
 @ ./node_modules/request/request.js
 @ ./node_modules/request/index.js
 @ ./src/cookie-maker-client.js
 @ ./node_modules/aurelia-webpack-plugin/runtime/empty-entry.js
 @ multi aurelia-webpack-plugin/runtime/empty-entry aurelia-webpack-plugin/runtime/pal-loader-entry (webpack)-dev-server/client?http://0.0.0.0:3000 webpack/hot/dev-server aurelia-bootstrapper

ERROR in ./node_modules/forever-agent/index.js
Module not found: Error: Can't resolve 'net' in '/project/cookie-trader/client/node_modules/forever-agent'
 @ ./node_modules/forever-agent/index.js 6:10-24
 @ ./node_modules/request/request.js
 @ ./node_modules/request/index.js
 @ ./src/cookie-maker-client.js
 @ ./node_modules/aurelia-webpack-plugin/runtime/empty-entry.js
 @ multi aurelia-webpack-plugin/runtime/empty-entry aurelia-webpack-plugin/runtime/pal-loader-entry (webpack)-dev-server/client?http://0.0.0.0:3000 webpack/hot/dev-server aurelia-bootstrapper

ERROR in ./node_modules/tough-cookie/lib/cookie.js
Module not found: Error: Can't resolve 'net' in '/project/cookie-trader/client/node_modules/tough-cookie/lib'
 @ ./node_modules/tough-cookie/lib/cookie.js 32:10-24
 @ ./node_modules/request/lib/cookies.js
 @ ./node_modules/request/index.js
 @ ./src/cookie-maker-client.js
 @ ./node_modules/aurelia-webpack-plugin/runtime/empty-entry.js
 @ multi aurelia-webpack-plugin/runtime/empty-entry aurelia-webpack-plugin/runtime/pal-loader-entry (webpack)-dev-server/client?http://0.0.0.0:3000 webpack/hot/dev-server aurelia-bootstrapper

ERROR in ./node_modules/tunnel-agent/index.js
Module not found: Error: Can't resolve 'net' in '/project/cookie-trader/client/node_modules/tunnel-agent'
 @ ./node_modules/tunnel-agent/index.js 3:10-24
 @ ./node_modules/request/lib/tunnel.js
 @ ./node_modules/request/request.js
 @ ./node_modules/request/index.js
 @ ./src/cookie-maker-client.js
 @ ./node_modules/aurelia-webpack-plugin/runtime/empty-entry.js
 @ multi aurelia-webpack-plugin/runtime/empty-entry aurelia-webpack-plugin/runtime/pal-loader-entry (webpack)-dev-server/client?http://0.0.0.0:3000 webpack/hot/dev-server aurelia-bootstrapper

ERROR in ./node_modules/forever-agent/index.js
Module not found: Error: Can't resolve 'tls' in '/project/cookie-trader/client/node_modules/forever-agent'
 @ ./node_modules/forever-agent/index.js 7:10-24
 @ ./node_modules/request/request.js
 @ ./node_modules/request/index.js
 @ ./src/cookie-maker-client.js
 @ ./node_modules/aurelia-webpack-plugin/runtime/empty-entry.js
 @ multi aurelia-webpack-plugin/runtime/empty-entry aurelia-webpack-plugin/runtime/pal-loader-entry (webpack)-dev-server/client?http://0.0.0.0:3000 webpack/hot/dev-server aurelia-bootstrapper

ERROR in ./node_modules/tunnel-agent/index.js
Module not found: Error: Can't resolve 'tls' in '/project/cookie-trader/client/node_modules/tunnel-agent'
 @ ./node_modules/tunnel-agent/index.js 4:10-24
 @ ./node_modules/request/lib/tunnel.js
 @ ./node_modules/request/request.js
 @ ./node_modules/request/index.js
 @ ./src/cookie-maker-client.js
 @ ./node_modules/aurelia-webpack-plugin/runtime/empty-entry.js
 @ multi aurelia-webpack-plugin/runtime/empty-entry aurelia-webpack-plugin/runtime/pal-loader-entry (webpack)-dev-server/client?http://0.0.0.0:3000 webpack/hot/dev-server aurelia-bootstrapper
webpack: Failed to compile.
```

Reply:
Looks like it requires some extra configuration. This thread details how to use `request` in the browser: https://github.com/request/request/issues/1529

I think this gets to a larger question though. Is our JS example code meant to be run from the browser or from Node.js? The JS SDK only works out of the box with Node.js, requiring some extra translation from a library like webpack to run in the browser. Furthermore, certain parts of the SDK will _never_ work in the browser. With this in mind, any example code I wrote was written for Node.js.

I understand the desire to write example code that "just works", but considering the wide variety of environments out there, I don't think we'll ever be able to guarantee that. Considering that large portions of the JS SDK do not work at all in the browser, I think we should keep the example code consistent and target Node.js.

Reply:
It is news to me the Javascript SDK is intended for use in the node only environment. Since I just pulled example usage code from browser components in the marketplace and supply chain examples. If the Javascript SDK is intended to be Node.js only that should be called out on first page of the documentation. It should probably be presented as we have a "Node.js Javascript SDK" not as a Javascript SDK.

Reply:
I would not classify it as "Node.js only" (though the processor components of it are). But it certainly only works out of the box in Node.js. It is usable from the browser, but to really support the browser as a first-class platform, we need to do a little more work. For example, we should provide the client code minified with dependencies, so it can be imported without webpack.

Anyway, hope your drive was otherwise enjoyable. Have a great last day, and good luck with your next project. The PR looks good to me otherwise.